vulcanai2/models/metrics.py

```diff
@@ -1,32 +1,66 @@
-__author__ = 'Caitrin'
-"""
-this will define any metrics like average etc, and reutrn
-
-"""
 import numpy as np
 import torch
 import torch.nn.functional as F
 from torch.autograd import Variable
 from sklearn import metrics as scipy_metrics
+import logging
+logger = logging.getLogger(__name__)
 
 class Metrics(object):
 
+    def __init__(self, num_class):
+        self.num_class = num_class
+        self.confusion_matrix = np.zeros((num_class, num_class))
+
+    def _fast_hist(self, prediction, target, num_class):
+        mask = (target >= 0) & (target < num_class)
+        hist = np.bincount(
+                            num_class * target[mask].astype(int) + prediction[mask],
+                            minlength=num_class ** 2,
+                            ).reshape(num_class, num_class)
+        return hist
+    def update(self, predictions, targets):
+        for prediction, target in zip(predictions, targets):
+            self.confusion_matrix += self._fast_hist(
```

Comment:
you can just use sklearn to generate the confusion matrix

Reply:
@priyatharsan I think this is relevant to you

Reply:
yup fixed it